src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.MemoryMappedInfo.cs

```diff
@@ -124,11 +120,7 @@ public Stream CreateReadableStream()
             /// </summary>
             public Stream CreateWritableStream()
             {
-                // CreateViewStream is not guaranteed to be thread-safe
-                lock (_memoryMappedFile.Target)
-                {
-                    return RunWithCompactingGCFallback(info => info._memoryMappedFile.Target.CreateViewStream(info.Offset, info.Size, MemoryMappedFileAccess.Write), this);
-                }
+                return RunWithCompactingGCFallback(info => info._memoryMappedFile.Target.CreateViewStream(info.Offset, info.Size, MemoryMappedFileAccess.Write), this);
```

Comment:
CreateReadableStream seems fine.  I'm having trouble convinging myself that this side is fine though.

Reply:
The implementations side is exactly the same. Is there a specific concern?